### ISSUE TITLE
RFE: collect database operations

### ISSUE THREAD

**Issue by stiller** [drogus/bulk_api#10]
A single bulk call on multiple items still results in multiple db operations. Perhaps these could be bundles together somehow. (Example: marking all as done or clearing al marked-as-done in todos app)

**Comment by drogus** [comment]
I agree that this could be done better, but there is no easy mechanism for that in Rails. Basically when you have validations and callbacks you can't just always update multiple items with one query. I guess we could have some kind of API to allow run specific methods in certain situations (like: mark_as_completed, which will mark multiple items as completed with one SQL query), but I don't have idea how could it look like. If you have any ideas on that, feel free to comment and/or send patch :)
